src/geomData.js

```diff
@@ -604,11 +619,30 @@ vgl.sourceData = function(arg) {
     var i;
 
     //m_data = m_data.concat(data); //no, slow on Safari
+    /* If we will are given a Float32Array and don't have any other data, use
+     * it directly */
+    if (!m_data.length && data.length &&
+        Object.prototype.toString.call(data) === "[object Float32Array]") {
```

Comment:
No.  `typeof` of a `Float32Array` returns `object`.  There might be a better way than what I used, but I haven't found it.